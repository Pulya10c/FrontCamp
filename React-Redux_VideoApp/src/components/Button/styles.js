import styled from 'styled-components';

export const Button = styled.button`
  font-family: 'Open Sans', sans-serif;
  font-weight: 800;
  font-size: 18px;
  line-height: 24px;
  outline: none;
  background-color: #004db9;
  border: 0;
  border-radius: 4px;
  color: white;
  padding: 15px;
  cursor: pointer;
  transition: all 0.3s cubic-bezier(0.075, 0.82, 0.165, 1);

  &:disabled {
    cursor: not-allowed;
    background-color: #7792b8;
  }

  &:hover:not(:disabled) {
    background-color: #00183a;
  }

  &:active {
    background-color: #004db9;
    opacity: 0.7;
  }
`;
