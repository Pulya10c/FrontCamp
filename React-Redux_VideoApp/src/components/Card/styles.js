import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const CardStyled = styled(Link)`
  display: grid;
  grid-gap: 12px;
  align-content: start;
  padding: 4px;
  font-family: 'Open Sans', sans-serif;
  border-radius: 10px;
  background-color: #fff;
  border: 1px solid rgba(218, 220, 224, 0.5);
  box-shadow: 0 20px 20px -20px $main-color;
  transition: all 0.3s cubic-bezier(0.075, 0.82, 0.165, 1);
  cursor: pointer;
  &:hover {
    transform: scale(1.02);
    box-shadow: 0 2px 2px #00183a;
  }
`;
