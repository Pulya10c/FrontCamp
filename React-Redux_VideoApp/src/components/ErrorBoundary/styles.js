import styled, { css } from 'styled-components'

export const commonAnimation = css`
  opacity: 0;
  animation: appear 500ms cubic-bezier(0, 1, 0.5, 1) 1 normal forwards;

  @keyframes appear {
    0% {
      opacity: 0;
      transform: translateY(100%);
    }
    100% {
      opacity: 1;
      transform: translateY(0);
    }
  }
`

export const ErrorWrapper = styled.div`
  margin: 40px auto;
  text-align: center;
`

export const ErrorTitle = styled.h1`
  ${commonAnimation}

  font-size: 40px;
  animation-delay: 0;
`

export const ErrorSubTitle = styled.h2`
  ${commonAnimation}

  font-size: 30px;
  animation-delay: 200ms;
`

export const ErrorContent = styled.p`
  ${commonAnimation}

  font-size: 18px;
  margin: 2em;
  animation-delay: 500ms;
`
