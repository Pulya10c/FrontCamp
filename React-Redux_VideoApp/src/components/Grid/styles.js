import styled from 'styled-components';
import { mediaMd, mediaSm } from 'utils/style-utils';

export const Grid = styled.section`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 5px;
  font-size: 20px;
  padding: 5px;

  ${mediaMd`
    grid-template-columns: repeat(3, 1fr);
    font-size: 24px;
  `};

  ${mediaSm`
    grid-template-columns: repeat(1, 1fr);
    font-size: 20px;
  `};
`;
