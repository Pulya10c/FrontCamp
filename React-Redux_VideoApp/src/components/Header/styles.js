import styled from 'styled-components';
import blurImage from './blurImage.jpg';

export const Header = styled.header`
  display: grid;
  grid-gap: 10px;
  justify-content: center;
  align-content: center;

  padding: 10px;
  min-height: 200px;
  background-image: url(${blurImage});
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
`;
