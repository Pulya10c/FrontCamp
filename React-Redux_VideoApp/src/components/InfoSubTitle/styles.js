import styled from 'styled-components';
import { mediaMd, mediaSm } from 'utils/style-utils';

export const InfoSubTitle = styled.h5`
  font-size: 16px;
  color: white;

  ${mediaMd`
    font-size: 14px;
  `};

  ${mediaSm`
    font-size: 12px;
  `};
`;
