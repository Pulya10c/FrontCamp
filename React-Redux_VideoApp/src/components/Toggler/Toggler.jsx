import React, { Component } from "react";
import PropTypes from "prop-types";

import {
  TogglerWrapper,
  TogglerRadioOption,
  TogglerLabelOption,
  TogglerTitle
} from "./styles";

class Toggler extends Component {
  // static propTypes = {
  //   prop: PropTypes
  // }

  render() {
    const { togglerTitle, selectedValue, togglerOptions } = this.props;

    return (
      <TogglerWrapper>
        <TogglerTitle>{togglerTitle}</TogglerTitle>
        {togglerOptions.map((option, i) => (
          <React.Fragment key={`${option}${i}`}>
            <TogglerRadioOption
              type="radio"
              name={togglerTitle}
              id={`${togglerTitle}${i}`}
              value={option}
              checked={selectedValue === option}
              onChange={this.handleOptionChange}
            />
            <TogglerLabelOption htmlFor={`${togglerTitle}${i}`}>
              {option}
            </TogglerLabelOption>
          </React.Fragment>
        ))}
      </TogglerWrapper>
    );
  }

  handleOptionChange = ({ target: { value } }) => {
    const { setToggleValue } = this.props;
    setToggleValue(value);
  };
}

export default Toggler;
