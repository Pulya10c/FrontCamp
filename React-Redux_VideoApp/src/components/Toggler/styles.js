import styled, { css } from 'styled-components';

export const TogglerWrapper = styled.div`
  display: grid;
  grid-gap: 5px;
  grid-auto-flow: rows;
  grid-template-columns: auto;
  justify-content: center;
  align-content: center;
`;

export const TogglerRadioOption = styled.input`
  visibility: hidden;
  height: 0;
  width: 0;

  &:checked + label {
    background-color: #58ba83;
  }
`;

export const TogglerLabelOption = styled.label`
  display: table-cell;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  background-color: #454545;
  color: white;
  padding: 5px 10px;
  border-radius: 3px;
  transition: all 0.3s ease-out;
`;

export const TogglerTitle = styled.h2`
  font-size: 20px;
`;
