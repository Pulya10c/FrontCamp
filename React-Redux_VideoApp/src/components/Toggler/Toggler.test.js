import React from 'react';
import { shallow } from 'enzyme';
import Toggler from './Toggler';
import { TogglerRadioOption } from './styles';

describe('Component Toggler', () => {
  let wrapper, defaultProps;

  beforeEach(() => {
    defaultProps = {
      togglerTitle: 'mockTitle',
      selectedValue: 'mockSelectedValue',
      togglerOptions: [
        { code: 'code1', text: 'text1' },
        { code: 'code2', text: 'text2' }
      ],
      setToggleValue: jest.fn()
    };
    wrapper = shallow(<Toggler {...defaultProps} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call defaultProps.setToggleValue with correct args if call TogglerRadioOption onChange', () => {
    const mockVal = 'mockVal';

    wrapper
      .find(TogglerRadioOption)
      .first()
      .props()
      .onChange({ target: { value: mockVal } });
    expect(defaultProps.setToggleValue).toHaveBeenCalledWith(mockVal);
  });
});
