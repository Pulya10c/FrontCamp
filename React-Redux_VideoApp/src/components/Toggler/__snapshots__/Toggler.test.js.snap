// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Toggler should render correctly 1`] = `
<styled.div>
  <styled.h2>
    mockTitle
  </styled.h2>
  <styled.input
    checked={false}
    id="code10"
    name="code1"
    onChange={[Function]}
    type="radio"
    value="code1"
  />
  <styled.label
    htmlFor="code10"
  >
    text1
  </styled.label>
  <styled.input
    checked={false}
    id="code21"
    name="code2"
    onChange={[Function]}
    type="radio"
    value="code2"
  />
  <styled.label
    htmlFor="code21"
  >
    text2
  </styled.label>
</styled.div>
`;
